docs/DeveloperGuide.md

```diff
@@ -1,29 +1,165 @@
 # Developer Guide
 
+* [Acknowledgements](#acknowledgements)
+* [Setting up, getting started] (#setting-up-getting-started)
+* [Design & implementation](#design--implementation)
+  * [Architecture](#architecture)
+  * [Category feature](#category-feature)
+* [Product scope](#product-scope)
+  * [Target user profile](#target-user-profile)
+  * [Value proposition](#value-proposition)
+* [User Stories](#user-stories)
+* [Non-Functional Requirements](#non-functional-requirements)
+* [Glossary](#glossary)
+* [Instructions for manual testing](#instructions-for-manual-testing)
+
 ## Acknowledgements
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
+## Setting up, getting started
+
+Refer to the guide [Setting up and getting started]().
+
 ## Design & implementation
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Architecture
+
+The UML diagram below shows the main relationships between the classes in the Brokeculator application.
+![img.png](images/architecture.png)
+
+### Category feature
+**Implementation** </br>
+
+The category feature is mainly facilitated by the `Category` class. The `Category` class is responsible for storing the names of the categories present in expenses. 
+In order for the user to be able to add expenses with a category, the category must be added using the `addCategory` method. 
+The `addCategory` method takes in a string as a parameter and adds it to the set of categories.
+The `Category` class implements the following operations:
+- `addCategory(String category)` - Adds a category to the set of categories
+- `getCategoryListString()` - Returns a string representation of the set of categories
+- `removeCategory(String category)` - Deletes a category from the set of categories
+
+The `Category` class is supplemented by the following classes to interact with the user:
+- `CategoryCommand` - This class is responsible for handling and executing the commands related to categories 
+- `CategoryParser` It is responsible for parsing the user input
+
+The UML diagram below shows the main relationships between the classes in the category feature.
+![category_class.png](images/category_class.png)
+</br>
```

Comment:
Use of line break can be done with `\` or `<br>`. This is currently being rendered in the html version of the DG.